internal/graph/check.go

```diff
@@ -355,6 +358,8 @@ func (c *LocalChecker) dispatch(ctx context.Context, req *ResolveCheckRequest) C
 	}
 }
 
+// ResolveCheck resolves a node out of a tree of evaluations. If the depth of the tree has gotten too large,
```

Comment:
nit - I'd like us to stop using "node" where possible. It's not at all clear what it refers to within the Zanzibar domain. Can we start referring to these as graph "subproblems" instead?

> ResolveCheck resolves a single graph subproblem out of the larger Check resolution.

Reply:
> It's not at all clear what it refers to within the Zanzibar domain

I was using graph nomenclature (node, edge, etc) since this in the graph package

Reply:
We don't really even agree what a "node" represents though. We have no visual representation for it at this time. That's why I mention it. That's why I think "graph subproblem" is more appropriate as it avoids the term node which doesn't have a concrete representation right now.

How do others feel about this common nomenclature? I'd like us to start converging on some terms so it's easier to communicate these things both internally and externally.

@adriantam @rhamzeh 

Maybe we can just avoid graph nomenclature altogether though. Instead say something like

> ResolveCheck resolve a single evaluation path out of a tree of evaluation paths. ....

Reply:
I don't know if PR is the best avenue to converge on terms. Let's take this discussion outside of this PR

Reply:
If anything i think the doc I added is better than having nothing 😆

Reply:
Fair point. My motivation was just trying to get us to have common vernacular, not because I care so much about `internal` godoc strings.